bigframes/core/blocks.py

```diff
@@ -1080,33 +1080,33 @@ def aggregate_all_and_stack(
             # TODO: Allow to promote identity/total_order columns instead for better perf
             offset_col = guid.generate_guid()
             expr_with_offsets = self.expr.promote_offsets(offset_col)
-            stacked_expr = expr_with_offsets.unpivot(
-                row_labels=self.column_labels.to_list(),
-                index_col_ids=[guid.generate_guid()],
-                unpivot_columns=[(value_col_id, tuple(self.value_columns))],
+            stacked_expr, (_, value_col_ids, passthrough_cols,) = unpivot(
+                expr_with_offsets,
+                row_labels=self.column_labels,
+                unpivot_columns=[tuple(self.value_columns)],
                 passthrough_columns=[*self.index_columns, offset_col],
             )
             index_aggregations = [
                 (ex.UnaryAggregation(agg_ops.AnyValueOp(), ex.free_var(col_id)), col_id)
-                for col_id in [*self.index_columns]
+                for col_id in passthrough_cols[:-1]
```

Comment:
Why every column except the last one? Could you have a comment here explaining, please?

Reply:
Added comment. These correspond to the passthrough_columns argument in unpivot.